### ISSUE TITLE
Players can take items from the creative inventory without the creative priv

### ISSUE THREAD

**Issue by elinor-s** [minetest/minetest_game#1839]
The detached creative inventory is created even for players who don't have the creative priv and [the code](https://github.com/minetest/minetest_game/blob/0.4.16/mods/creative/inventory.lua#L11) doesn't check when a player tries to take items that the player has the creative priv. So any player can take items from the creative inventory on a public server. With `minetest.show_formspec`, this doesn't even require modifying the client:

```lua
local current_page = 0

local function show_creative_inventory(page)
	if page < 0 then
		page = 0
	end
	current_page = page
	local player_name = minetest.localplayer:get_name()
	local formspec = ([[
			size[8,8.5]
			label[6.3,3.35;Page %d]
			image[4.06,3.4;0.8,0.8;creative_trash_icon.png]
			listcolors[#00000069;#5A5A5A;#141318;#30434C;#FFF]
			list[current_player;main;0,4.7;8,1;]
			list[current_player;main;0,5.85;8,3;8]
			list[detached:creative_trash;main;4,3.3;1,1;]
			listring[]
			button[5.4,3.2;0.8,0.9;creative_prev;<]
			button[7.25,3.2;0.8,0.9;creative_next;>]
			listring[current_player;main]
			listring[detached:creative_%s;main]
			list[detached:creative_%s;main;0,0;8,3;%d]
		]]):format(page+1, player_name, player_name, current_page*8*3)
	minetest.show_formspec("creative_inventory", formspec)
end

minetest.register_on_formspec_input(function(formname, fields)
	if formname == "creative_inventory" then
		if fields.creative_prev then
			show_creative_inventory(current_page - 1)
		elseif fields.creative_next then
			show_creative_inventory(current_page + 1)
		end
	end
end)

minetest.register_chatcommand("creative_inventory", {
	params = "",
	description = "Show the creative inventory",
	func = function(param)
		show_creative_inventory(current_page)
	end,
})
```

Server owners should consider fixing this as soon as possible.

**Comment by ForbiddenJ** [comment]
It looks like what you're saying is true. The init.lua file [executes](https://github.com/minetest/minetest_game/blob/0.4.16/mods/creative/init.lua#L9) the inventory file regardless of the global creative mode setting. The file in turn [executes an update](https://github.com/minetest/minetest_game/blob/0.4.16/mods/creative/init.lua#L9) to an assigned creative inventory regardless of the player actually having any sort of privilege, creating a detached inventory assigned to the player. Finally there is [no check in place](https://github.com/minetest/minetest_game/blob/0.4.16/mods/creative/inventory.lua#L23) for prevent unauthorized obtaining of materials.

To make things worse, CSM can be used to get free anything on most servers with this bug.

**Comment by ExeterDad** [comment]
@elinor-s Thanks for posting this. I came here to file a report as I heard you reported it on our server.

**Comment by Fixer-007** [comment]
This will be in 0.4.17 too, right?

**Comment by sfan5** [comment]
As you can see [here](https://github.com/minetest/minetest_game/projects) it was already backported into the respective branch.
TL;DR **yes**
